pkg/v1/tkg/client/upgrade_addon.go

```diff
@@ -327,14 +327,17 @@ func (c *TkgClient) setProxyConfiguration(clusterClusterClient clusterclient.Cli
 
 	if httpProxy := configmap.Data["httpProxy"]; httpProxy != "" {
 		c.TKGConfigReaderWriter().Set(constants.TKGHTTPProxy, httpProxy)
+		c.TKGConfigReaderWriter().Set(constants.TKGHTTPProxyEnabled, trueString)
 	}
 
 	if httpsProxy := configmap.Data["httpsProxy"]; httpsProxy != "" {
 		c.TKGConfigReaderWriter().Set(constants.TKGHTTPSProxy, httpsProxy)
+		c.TKGConfigReaderWriter().Set(constants.TKGHTTPProxyEnabled, trueString)
 	}
 
 	if noProxy := configmap.Data["noProxy"]; noProxy != "" {
 		c.TKGConfigReaderWriter().Set(constants.TKGNoProxy, noProxy)
+		c.TKGConfigReaderWriter().Set(constants.TKGHTTPProxyEnabled, trueString)
```

Comment:
I don't think you need to set `TKGHTTPProxyEnabled` when checking for noproxy. If httpproxy and httpsproxy doesn't exist, proxy is not enabled. Strickly speaking, httpproxy and httpsproxy have to exist at the same time, so checking only httpproxy is sufficient to determine whether `TKGHTTPProxyEnabled` need to be set or not

Reply:
Thanks. Updated.